cranelift-codegen/shared/src/lib.rs

```diff
@@ -19,6 +19,15 @@
         clippy::use_self
     )
 )]
+#![no_std]
+
+#[allow(unused_imports)] // #[macro_use] is required for no_std
+#[macro_use]
+extern crate alloc;
+
+#[cfg(feature = "std")]
+#[macro_use]
+extern crate std;
```

Comment:
It should be possible to never use std in cranelift_codegen_shared, right?

Reply:
Right now, yes. There's one unit test that uses std, and it can be replaced with alloc. I can get rid of that if you want; you'll just have to add it back if you write another unit test or some feature-gated code that requires it.